assets/src/edit-story/components/library/karma/shapes/shapes.karma.js

```diff
@@ -0,0 +1,106 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/**
+ * External dependencies
+ */
+import { act } from '@testing-library/react';
+
+/**
+ * Internal dependencies
+ */
+import { Fixture } from '../../../../karma/fixture';
+
+describe('TextEdit integration', () => {
+  let fixture;
+
+  beforeEach(async () => {
+    fixture = new Fixture();
+
+    await fixture.render();
+  });
+
+  afterEach(() => {
+    fixture.restore();
+  });
+  //d="M 0.5 0 L 1 1 L 0 1 Z" triangle.
+
+  it('should render ok', () => {
+    expect(
+      fixture.container.querySelector('[data-testid="fullbleed"]')
+    ).toBeTruthy();
+  });
+
+  it('via clicking on shape preview', async () => {
+    let shapes = await fixture.container.querySelectorAll(
+      '[data-testid="safezone"] [d="M 0.5 0 L 1 1 L 0 1 Z"]'
+    );
+    expect(shapes.length).toBe(0);
+
+    //await karmaPause();
+    await act(async () => {
+      await fixture.events.click(
+        fixture.querySelector(`[id="library-tab-shapes"]`)
+      );
+      await fixture.events.click(
+        fixture.querySelector('[aria-label="Triangle"]')
+      );
+
+      shapes = await fixture.container.querySelectorAll(
+        '[data-testid="safezone"] [d="M 0.5 0 L 1 1 L 0 1 Z"]'
+      );
+    });
+    expect(shapes.length).toBeGreaterThan(0);
+  });
+
+  it('via dragging from shape preview', async () => {
+    let shapes = await fixture.container.querySelectorAll(
+      '[data-testid="safezone"] [d="M 0.5 0 L 1 1 L 0 1 Z"]'
+    );
+    expect(shapes.length).toBe(0);
+
+    await act(async () => {
+      await fixture.events.click(
+        fixture.querySelector(`[id="library-tab-shapes"]`)
+      );
+      // Move the mouse to the middel of the shape preview and start mouse press
+      // down.
+      const shapePreview = await fixture.querySelector(
+        '[aria-label="Triangle"]'
+      );
+      const shapeBoundingBox = await shapePreview.getBoundingClientRect();
+      await fixture.events.mouse.move(
+        shapeBoundingBox.x + shapeBoundingBox.width / 2,
+        shapeBoundingBox.y + shapeBoundingBox.height / 2
+      );
+      await fixture.events.mouse.down();
+
+      // Move the mouse to the middle of the safe area and then mouse press up.
+      const safezone = await fixture.querySelector('[data-testid="safezone"]');
+      const safezoneBox = await safezone.getBoundingClientRect();
+      await fixture.events.mouse.move(
+        safezoneBox.x + safezoneBox.width / 2,
+        safezoneBox.y + safezoneBox.height / 2
+      );
+
+      await fixture.events.mouse.up();
+    });
+
+    shapes = await fixture.container.querySelectorAll(
+      '[data-testid="safezone"] [d="M 0.5 0 L 1 1 L 0 1 Z"]'
+    );
+    expect(shapes.length).toBeGreaterThan(0);
```

Comment:
So, [the test fails on this line](https://github.com/google/web-stories-wp/pull/1856/checks?check_run_id=733349457#step:7:35) - `shapes.length` is apparently `0` at this point. Does it work locally for you?

Reply:
Hey it fails as well for me, sorry I didn't clarify. It seems to pick up the shapePreview triangle but not drop it safe area properly (and it tries to drop it where my mouse cursor is in real life rather than in the puppeteer window).

I was uploading this to see if it worked on the pr instead.

Reply:
As per the slack thread, I will xit() the test so that it skips.